plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1788,6 +1797,23 @@ else if (partitionUpdate.getUpdateMode() == NEW || partitionUpdate.getUpdateMode
                         .collect(toImmutableList())));
     }
 
+    private void removeNonCurrentQueryFiles(ConnectorSession session, Path partitionPath)
+    {
+        try {
+            FileSystem fileSystem = hdfsEnvironment.getFileSystem(new HdfsContext(session), partitionPath);
+            RemoteIterator<LocatedFileStatus> iterator = fileSystem.listFiles(partitionPath, false);
+            while (iterator.hasNext()) {
+                Path file = iterator.next().getPath();
+                if (!HiveWriteUtils.isFileCreatedByQuery(file, session)) {
```

Comment:
Pass just what's required - `file.getName()` and `session.getQueryId()` - would no longer need the overloaded function (since it doesn't do anything with the session anyway).